### PR TITLE
feat: iOS Auto-Layout Yoga interworking

### PR DESCRIPTION
**Pull request by dantes-git**
### Related Issues

#810

### Description and Example

- implements _intrinsicContentSize_ in `BeagleView`
- implements a easy way of using a AutoLayout Component inside Beagle through a wrapper (`AutoLayoutWrapper`)

Sample:
```swift
class AutoLayoutSample: UIView {
    // impl of a view using AutoLayout
}

struct AutoLayoutComponent: ServerDrivenComponent {
    var widgetProperties: WidgetProperties = WidgetProperties()
    
    func toView(renderer: BeagleRenderer) -> UIView {
        return AutoLayoutWrapper(view: AutoLayoutSample())
    }
}
```
### Checklist

Please, check if these important points are met using `[x]`:

- [x] I read the [PR Guide] and followed the process outlined there for submitting this PR.
- [x] I avoided _breaking changes_ by not changing public APIs that people rely on. 
- [x] I am willing to follow-up on review comments in a timely manner.

[PR Guide]: https://github.com/ZupIT/beagle/blob/master/doc/contributing/pull_requests.md

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: iOS/Sources/Beagle/Sources/Renderer/BeagleView.swift [hunk 1]
```diff
@@ -48,6 +50,7 @@ public class BeagleView: UIView {
     // MARK: - Lifecycle
     
     public override func didMoveToWindow() {
+        super.didMoveToWindow()
```
Comment (caioortu) [inline iOS/Sources/Beagle/Sources/Renderer/BeagleView.swift:53]: This method has no default implementation, we don't need to call it.

File: iOS/Sources/Beagle/Sources/Renderer/BeagleScreenViewController.swift [hunk 2]
```diff
@@ -48,6 +48,9 @@ public class BeagleScreenViewController: BeagleController {
     
     private var navigationControllerId: String?
     
+    // TODO: This workaround should be removed in BeagleView future implementation
```
Comment (botbeaglezup) [inline iOS/Sources/Beagle/Sources/Renderer/BeagleScreenViewController.swift:51]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/ac7da0e83c11ceecc17470629a63d5b52758bad1/iOS/Sources/Beagle/Sources/Renderer/BeagleScreenViewController.swift#L51"></span>TODOs should be resolved (This workaround should be remo…).<br />
<code>todo</code> <code>BeagleScreenViewController.swift:51</code></td>
    </tr>
  </tbody>
</table>

**Review by lucasfonsecazup — APPROVED** [review]

**Comment by theffc** [comment]
I was testing manually BeagleDemo, and in the PageView screen I saw something that could have happened due to these changes:
- The remote image inside that screen is being inconsistently shown. For some pages it appears and in others it doesn't.

Correct behavior for all pages:
Simulator Screen Shot - iPhone 11 - 2020-09-21 at 13 50 36

Wrong behavior in some pages:
Simulator Screen Shot - iPhone 11 - 2020-09-21 at 13 50 40

File: iOS/Sources/Beagle/Sources/Renderer/BeagleScreenViewController.swift [hunk 3]
```diff
@@ -48,6 +50,9 @@ public class BeagleScreenViewController: BeagleController {
...
```
Comment (botbeaglezup) [inline iOS/Sources/Beagle/Sources/Renderer/BeagleScreenViewController.swift:53]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/2bef07259e5779a36068c76e4fc8bed9d8a2f636/iOS/Sources/Beagle/Sources/Renderer/BeagleScreenViewController.swift#L53"></span>TODOs should be resolved (This workaround should be remo…).<br />
<code>todo</code> <code>BeagleScreenViewController.swift:53</code></td>
    </tr>
  </tbody>
</table>

**Review by theffc — APPROVED** [review]

**Review by caioortu — DISMISSED** [review]

**Review by lucasaraujo — APPROVED** [review]

**Review by theffc — APPROVED** [review]

**Review by gabcoelho — APPROVED** [review]

**Review by luisgustavozup — APPROVED** [review]

**Review by caioortu — APPROVED** [review]

**Pull request merged**